src/stats.rs

```diff
@@ -262,32 +281,45 @@ impl GooseStats {
                     aggregate_min_response_time,
                     aggregate_max_response_time
                 ),
-            );
+            )?;
         }
 
-        self
+        Ok(())
     }
 
-    // Display slowest response times within several percentiles.
-    pub fn print_percentiles(&self) -> &GooseStats {
+    // Optionallyl prepares a table of slowest response times within several percentiles.
+    pub fn fmt_percentiles(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         // If there's nothing to display, exit immediately.
-        if self.requests.is_empty() {
-            return self;
+        if !self.display_percentile {
+            return Ok(());
         }
 
         let mut aggregate_response_times: BTreeMap<usize, usize> = BTreeMap::new();
         let mut aggregate_total_response_time: usize = 0;
         let mut aggregate_response_time_counter: usize = 0;
         let mut aggregate_min_response_time: usize = 0;
         let mut aggregate_max_response_time: usize = 0;
-        println!("-------------------------------------------------------------------------------");
-        println!(" Slowest page load within specified percentile of requests (in ms):");
-        println!(" ------------------------------------------------------------------------------");
-        println!(
+        writeln!(
+            f,
+            "-------------------------------------------------------------------------------"
+        )?;
+        writeln!(
+            f,
+            " Slowest page load within specified percentile of requests (in ms):"
+        )?;
+        writeln!(
+            f,
+            " ------------------------------------------------------------------------------"
+        )?;
+        writeln!(
```

Comment:
This is preference, but I feel:

```
writeln!(f,
```

is almost as short as:

```
println!(
```

so I would not split it over several lines in the whole PR as it makes things much more difficult to read.

Also the diff is shorter ;)

Reply:
I do not format white space manually, but instead use `cargo fmt`. We would need to set up custom rustfmt rules if we want to change how it formats things like this. (In this case I believe we'd need to adjust the line-length before a wrap). 

I prefer to stick with the defaults.

Reply:
I agree